modules/packages/Python.chpl

```diff
@@ -2217,6 +2217,157 @@ module Python {
     proc getPyObject() do return this.obj;
 
     // TODO: call should support kwargs
+
+    // Casts
+    /* Creates a new int from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: int(?)): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new uint from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: uint(?)): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new real from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: real(?)): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new bool from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: bool): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new string from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: string): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      // Check if this type is normally supposed to be a string.  To do that,
+      // get the normal Python string type.
+      var dummy = "s";
+      var dummyPyObj = x.interpreter.toPythonInner(dummy); // TODO: catch
+      var strType = Py_TYPE(dummyPyObj);
+      x.interpreter.checkException();
+
+      var pyObj = x.getPyObject();
+      var isStr = Py_IS_TYPE(pyObj, strType): bool;
+      x.interpreter.checkException();
```

Comment:
there is no need for this extra logic, you can use `PyUnicode_Check`

Reply:
I'm considering this resolved due to removing this particular overload